src/plugins/process/process.c

```diff
@@ -0,0 +1,243 @@
+			keyAddBaseName (childKey, "infos");
+			keyAddBaseName (childKey, "placements");
+
+			Key * childPlacementsKey = ksLookup (childReturned, childKey, KDB_O_NONE);
+			keySetString (placementsKey, keyString (childPlacementsKey));
+
+			ksAppend (returned, contract);
+			ksDel (contract);
+		}
+
+		keyDel (childKey);
+		ksDel (childReturned);
+		return ret;
+	}
+
+	// Otherwise business as usual
+	if (elektraPluginProcessIsParent (pp)) return elektraPluginProcessSend (pp, ELEKTRA_PLUGINPROCESS_GET, returned, parentKey);
+
+	int ret = elektraInvoke2Args (process->plugin, "get", returned, parentKey);
+	if (ret == -2) return ELEKTRA_PLUGIN_STATUS_NO_UPDATE;
+	return ret;
+}
+
+int elektraProcessSet (Plugin * handle, KeySet * returned, Key * parentKey)
+{
+	Process * process = elektraPluginGetData (handle);
+	ElektraPluginProcess * pp = process->pp;
+	if (elektraPluginProcessIsParent (pp)) return elektraPluginProcessSend (pp, ELEKTRA_PLUGINPROCESS_SET, returned, parentKey);
+
+	int ret = elektraInvoke2Args (process->plugin, "set", returned, parentKey);
+	if (ret == -2) return ELEKTRA_PLUGIN_STATUS_NO_UPDATE;
+	return ret;
+}
+
+int elektraProcessError (Plugin * handle, KeySet * returned, Key * parentKey)
+{
+	Process * process = elektraPluginGetData (handle);
+	ElektraPluginProcess * pp = process->pp;
+	if (elektraPluginProcessIsParent (pp)) return elektraPluginProcessSend (pp, ELEKTRA_PLUGINPROCESS_ERROR, returned, parentKey);
+
+	int ret = elektraInvoke2Args (process->plugin, "error", returned, parentKey);
+	if (ret == -2) return ELEKTRA_PLUGIN_STATUS_SUCCESS;
+	return ret;
+}
+
+int elektraProcessCheckConfig (Key * errorKey, KeySet * conf)
+{
+	// We need the plugin key to know which plugin we should proxy
+	Key * pluginNameKey = ksLookupByName (conf, "/plugin", KDB_O_NONE);
+	if (pluginNameKey == NULL)
+	{
+		ELEKTRA_SET_ERROR (192, errorKey, "Missing plugin configuration parameter plugin=<name of plugin to be proxied>");
+		return ELEKTRA_PLUGIN_STATUS_ERROR;
+	}
+
+	// and this key should obviously contain the plugin's name, so check for any name
+	// furthermore by invoking process in a process we'd create a deadloop, check that too
+	const char * pluginName = keyString (pluginNameKey);
+	if (elektraStrCmp (pluginName, "(null)") == 0 || elektraStrCmp (pluginName, "(binary)") == 0)
+	{
+		ELEKTRA_SET_ERRORF (192, errorKey, "Plugin configuration parameter plugin has an invalid value: %s", pluginName);
+		return ELEKTRA_PLUGIN_STATUS_ERROR;
+	}
+	else if (elektraStrCmp (pluginName, "process") == 0)
+	{
+		ELEKTRA_SET_ERROR (192, errorKey, "Cannot proxy the process plugin itself");
+		return ELEKTRA_PLUGIN_STATUS_ERROR;
+	}
+
+	return ELEKTRA_PLUGIN_STATUS_NO_UPDATE;
+}
+
+Plugin * ELEKTRA_PLUGIN_EXPORT (process)
+{
+	// clang-format off
+	return elektraPluginExport ("process",
+		ELEKTRA_PLUGIN_OPEN,	&elektraProcessOpen,
+		ELEKTRA_PLUGIN_CLOSE,	&elektraProcessClose,
+		ELEKTRA_PLUGIN_GET,		&elektraProcessGet,
```

Comment:
formatting?

Reply:
still not resolved

Reply:
funny, on my system it looks normal this way, but ok deleted one tab ;)

Reply:
Hopefully GitHub did not change their tab setting (or is different than ours)... I trusted the GitHub view and did not check locally. Our settings are in .clang-format: TabWidth is 8